src/app/[lang]/account/[id]/page.tsx

```diff
@@ -0,0 +1,48 @@
+import { getUser } from "@/data/webData"
+import { getActiveSession } from "@/lib/oauth"
+import { Locale, getDictionary } from "@/localization"
+import { cookies } from "next/headers"
+import { redirect } from "next/navigation"
+import { viewAnyAccountMinAccessLevel } from "@/lib/utils"
+import { PortableAccountPage } from "../portable-account-page"
+
+export default async function AnyAccountPage(
+    {
+        params,
+        searchParams
+    }: {
+        params: {
+            lang: Locale
+            id: string
+        },
+        searchParams: {
+            page: string
+        }
+    }
+) {
+
+    // get the current session
+    const session = await getActiveSession(cookies())
+
+    // redirect if the user is not able to view any account
+    if (!session || viewAnyAccountMinAccessLevel > session.user.accessLevel) return redirect("./")
+
+    // parse the user email
+    const user_email = `${params.id}@${process.env.GOOGLE_OAUTH_HD || ''}`
```

Comment:
The session doesn't contain the user email? Weird that google wouldn't provide that

Reply:
It does, it's just that only emails from murraystate.edu are allowed to create accounts/sign in, so there's no point in storing it in a session.